packages/kit/types.d.ts

```diff
@@ -1,3 +1,4 @@
+import './types.ambient';
```

Comment:
I don't understand what "ambient" means here. It means the types are just defined in a `.d.ts` file? But aren't the types in this file also ambient types?

Reply:
I meant ambient in the sense that you don't need to specifically import them, they are just available. I adopted the word "ambient" since the official TS docs also use it: https://www.typescriptlang.org/docs/handbook/modules.html#ambient-modules

Reply:
I'm not quite following why the types are split into two files and how it's decided what is in each file

Reply:
Everything in ambient needs to be in its own file because this file cannot have top level imports or it no longer works (this is how ts works, don't ask me why). The split between internal and public - I don't know.